cuML/src/glm/qn/qn_solvers.h

```diff
@@ -175,7 +176,8 @@ inline void update_pseudo(const SimpleVec<T> &x, const SimpleVec<T> &grad,
                           const int pg_limit, SimpleVec<T> &pseudo,
                           cudaStream_t stream) {
   if (grad.len > pg_limit) {
-    pseudo = grad;
+    pseudo.copy_async(grad, stream);
+    CUDA_CHECK(cudaStreamSynchronize(stream));
```

Comment:
Is the `copy_async` of `grad` to `pseudo` a device to device copy? If so I think the `cudaStreamSynchronize` is not really needed as subsequent operations are submitted in the same stream, right?

Reply:
See above